pkg/types/openstack/platform.go

```diff
@@ -62,6 +62,11 @@ type Platform struct {
 	// +optional
 	ClusterOSImage string `json:"clusterOSImage,omitempty"`
 
+	// ClusterOSImageProperties is a list of properties to be added to the metadata of the uploaded Glance ClusterOSImage.
+	// Default: the default is to not set any properties.
+	// +optional
+	ClusterOSImageProperties map[string]string `json:"clusterOSImageProperties,omitempty"`
```

Comment:
The description should include all of the properties and their values that we accept. At which point, we may as well specify them as separate fields rather than in a map.

Reply:
Done.

Reply:
So, now we are in a situation where there are only specific fields allowed in the map. This should not be a map but rather defined fields.
```suggestion
	ClusterOSImageProperties *ClusterOSImageProperties `json:"clusterOSImageProperties,omitempty"`
```
```
type ClusterOSImageProperties struct{
	QEMUGuestAgent QEMUGuestAgent `json:"qemuGuestAgent,omitempty"`
	DiskBus DiskBus `json:"diskBus,omitempty"`
	SCSIModel SCSIModel `json:"scsiModel,omitempty"`
}

// +kubebuilder:validation:Enum="";Enabled;Disabled
type QEMUGuestAgent string

const (
	EnabledQEMUGuestAgent QEMUGuestAgent = "Enabled"
	DisabledQEMUGuestAgent QEMUGuestAgent = "Disabled"
)

...
```

Reply:
I'd prefer we just expose glance images properties because this is the commonly accepted way to tune your VMs and these are already documented in openstack. Coming up with our own names is not good for discoverability once we start to support more of these properties.

Reply:
There is presedence for this elsewhere in the installer: 
https://github.com/openshift/installer/blob/master/pkg/types/aws/platform.go#L33

aws use of "userTags"

Reply:
One obstacle of using something like 

type ClusterOSImageProperties struct{
	QEMUGuestAgent QEMUGuestAgent `json:"qemuGuestAgent,omitempty"`
	DiskBus DiskBus `json:"diskBus,omitempty"`
	SCSIModel SCSIModel `json:"scsiModel,omitempty"`
}

Is that using enable/disable boolean to implement the UI, is that it becomes exponetially more dificult, becuse DiskBus, and SCSIModel are not booleans. They are not a disable/enable. They are fields that take arbritray values which are also mutually exclusive
so in reality it becomes like this: 

type ClusterOSImageProperties struct{
	QEMUGuestAgent QEMUGuestAgent `json:"qemuGuestAgent,omitempty"`
	DiskBus DiskBusSCSI `json:"diskBusSCSCI,omitempty"`
        DiskBus DiskBusVirtIO `json:"diskBusVirtIO,omitempty"`

	SCSIModel SCSIModel `json:"scsiModel,omitempty"`
}

And DiskBUsSCSI is now exclusive of DiskBUSVirtIO, (we have to code logic to make this so)
and they dependon the alue of SCSIMOdel, 
and it becomes much more complex to code and use and more likely to introduce bugs and user errors.

And I still don't see the actual value of obfuscating the options from the user.

Reply:
Perhaps the correct approach is to modify the validation. 
The real desire is to allow the user to provide any values they want, 
and as a precaution we inform the user if the field they provided is not in the list of supported (by openshift) options. 
But we do not really want to prevent the user, but our current validatio framework does not allow for warnings, only for errors.